server/middleware/validateRouteParams.js

```diff
@@ -1,116 +1,99 @@
-  check("boardId", "boardId is not defined").not().isEmpty(),
-  check("columnId", "columnId is not defined").not().isEmpty(),
+  handleParams("userId"),
+  handleParams("boardId"),
+  handleParams("columnId"),
   (req, res, next) => {
-    const errors = validationResult(req);
-
-    if (!errors.isEmpty())
-      return res.status(400).json({ errors: errors.array() });
-    next();
+    handleError(req, res, next);
   },
 ];
 
 exports.validateMoveCardOutParams = [
-  check("userId", "userId is not defined").not().isEmpty(),
-  check("boardId", "boardId is not defined").not().isEmpty(),
-  check("initialColumnId", "initialColumnId is not defined").not().isEmpty(),
-  check("targetColumnId", "targetColumnId is not defined").not().isEmpty(),
-  check("cardId", "cardId is not defined").not().isEmpty(),
-  check("targetPosition", "targetPosition is not defined").not().isEmpty(),
-  check("cardObject", "cardObject is not defined").not().isEmpty(),
+  handleParams("userId"),
+  handleParams("boardId"),
+  handleParams("initialColumnId"),
+  handleParams("targetColumnId"),
+  handleParams("cardId"),
+  handleParams("targetPosition"),
+  handleParams("cardObject"),
   (req, res, next) => {
-    console.log("valide 2");
-    const errors = validationResult(req);
-
-    if (!errors.isEmpty())
-      return res.status(400).json({ errors: errors.array() });
-    next();
+    handleError(req, res, next);
   },
 ];
 
 exports.validateMoveCardInParams = [
-  check("userId", "userId is not defined").not().isEmpty(),
-  check("boardId", "boardId is not defined").not().isEmpty(),
-  check("columnId", "columnId is not defined").not().isEmpty(),
-  check("cardId", "cardId is not defined").not().isEmpty(),
-  check("targetPosition", "targetPosition is not defined").not().isEmpty(),
-  check("cardObject", "cardObject is not defined").not().isEmpty(),
+  handleParams("userId"),
+  handleParams("boardId"),
+  handleParams("columnId"),
+  handleParams("cardId"),
+  handleParams("targetPosition"),
+  handleParams("cardObject"),
   (req, res, next) => {
-    const errors = validationResult(req);
-
-    if (!errors.isEmpty())
-      return res.status(400).json({ errors: errors.array() });
-    next();
+    handleError(req, res, next);
   },
 ];
 
-exports.valide = [
-  check("userId", "userId is not defined").not().isEmpty(),
-  check("boardId", "boardId is not defined").not().isEmpty(),
-  check("initialColumnId", "initialColumnId is not defined").not().isEmpty(),
-  check("targetColumnId", "targetColumnId is not defined").not().isEmpty(),
-  check("cardId", "cardId is not defined").not().isEmpty(),
-  check("targetPosition", "targetPosition is not defined").not().isEmpty(),
-  check("cardObject", "cardObject is not defined").not().isEmpty(),
+exports.validateCreateCardParams = [
+  handleParams("userId"),
+  handleParams("boardId"),
+  handleParams("cardTitle"),
+  handleParams("tagColor"),
+  handleParams("columnId"),
```

Comment:
Won't be needed as we are doing plugins which will handle validation on their controllers. 😄